crates/era-cheatcodes/tests/src/cheatcodes/ExpectRevert.t.sol

```diff
@@ -0,0 +1,246 @@
+        reverter.revertWithMessage("revert");
+    }
+
+    function testFailExpectRevertWrongString() public {
+         Reverter reverter = new Reverter();
+        (bool success, ) = Constants.CHEATCODE_ADDRESS.call(
+            abi.encodeWithSignature(
+                "expectRevert(bytes)",
+                "my not so cool error"));
+        require(success, "expectRevert failed");
+        reverter.revertWithMessage("my cool error");
+    }
+
+    function testExpectRevertCustomError() public {
+         Reverter reverter = new Reverter();
+        (bool success, ) = Constants.CHEATCODE_ADDRESS.call(
+            abi.encodeWithSignature(
+                "expectRevert(bytes)",
+                    abi.encodeWithSelector(Reverter.CustomError.selector, 1)));
+        require(success, "expectRevert failed");
+        reverter.revertWithCustomError(1);
+    }
+
+    function testFailExpectRevertCustomError() public {
+         Reverter reverter = new Reverter();
+        (bool success, ) = Constants.CHEATCODE_ADDRESS.call(
+            abi.encodeWithSignature(
+                "expectRevert(bytes)",
+                    abi.encodeWithSelector(Reverter.CustomError.selector, 1)));
+        require(success, "expectRevert failed");
+        reverter.revertWithCustomError(2);
+    }
+    // function testFailRevertNotOnImmediateNextCall() public {
+    //     Reverter reverter = new Reverter();
+    //     // expectRevert should only work for the next call. However,
+    //     // we do not immediately revert, so,
+    //     // we fail.
+    //     cheatcodes.expectRevert("revert");
+    //     reverter.doNotRevert();
+    //     reverter.revertWithMessage("revert");
+    // }
+
+    // function testExpectRevertConstructor() public {
+    //     cheatcodes.expectRevert("constructor revert");
+    //     new ConstructorReverter("constructor revert");
+    // }
+
+    // function testExpectRevertBuiltin() public {
+    //     Reverter reverter = new Reverter();
+    //     cheatcodes.expectRevert(abi.encodeWithSignature("Panic(uint256)", 0x11));
+    //     reverter.panic();
+    // }
+
+    // function testExpectRevertCustomError() public {
+    //     Reverter reverter = new Reverter();
+    //     cheatcodes.expectRevert(abi.encodePacked(Reverter.CustomError.selector));
+    //     reverter.revertWithCustomError();
+    // }
+
+    // function testExpectRevertNested() public {
+    //     Reverter reverter = new Reverter();
+    //     Reverter inner = new Reverter();
+    //     cheatcodes.expectRevert("nested revert");
+    //     reverter.nestedRevert(inner, "nested revert");
+    // }
+
+    // function testExpectRevertCallsThenReverts() public {
+    //     Reverter reverter = new Reverter();
+    //     Dummy dummy = new Dummy();
+    //     cheatcodes.expectRevert("called a function and then reverted");
+    //     reverter.callThenRevert(dummy, "called a function and then reverted");
+    // }
```

Comment:
Are these cases not yet supported?

Reply:
I assume @Karrq copypasted them from the cheatcodes and we will activate it, once we are supporting cheat codes calls directly

Reply:
@Karrq Can these be enabled (using the the long form call syntax)?

Reply:
Yes I did copy paste, and I simply had these removed for now not for the direct calls but to simplify a bit the tests honestly. These should be enabled (and fixed)